src/Entity/AuditRecord.php

```diff
@@ -109,6 +110,16 @@ public static function maintainerRemoved(Package $package, User $maintainer, ?Us
         return new self(AuditRecordType::MaintainerRemoved, ['name' => $package->getName(), 'maintainer' => self::getUserData($maintainer), 'actor' => self::getUserData($actor)], $actor?->getId(), $package->getVendor(), $package->getId(), $maintainer->getId());
     }
 
+    public static function packageAbandoned(Package $package, ?User $actor, ?string $replacementPackage, ?AbandonmentReason $reason = null): self
+    {
+        return new self(AuditRecordType::PackageAbandoned, ['name' => $package->getName(), 'repository' => $package->getRepository(), 'replacement_package' => $replacementPackage, 'reason' => $reason?->value, 'actor' => self::getUserData($actor, 'automation')], $actor?->getId(), $package->getVendor(), $package->getId());
+    }
+
+    public static function packageUnabandoned(Package $package, ?User $actor, ?string $previousReplacementPackage): self
```

Comment:
Do we really need to store the previous value? It should be in the previous audit log for that package marging the package abandoned, so seems a bit redundant.